atc/metric/emitter/prometheus.go

```diff
@@ -691,6 +693,21 @@ func (emitter *PrometheusEmitter) resourceMetric(logger lager.Logger, event metr
 	emitter.resourceChecksVec.WithLabelValues(team, pipeline).Inc()
 }
 
+func (emitter *PrometheusEmitter) checkMetric(logger lager.Logger, event metric.Event) {
+	scopeID, exists := event.Attributes["scope_id"]
+	if !exists {
+		logger.Error("failed-to-find-resource-config-scope-id-in-event", fmt.Errorf("expected scope_id to exist in event.Attributes"))
+		return
+	}
+	checkName, exists := event.Attributes["check_name"]
+	if !exists {
+		logger.Error("failed-to-find-check-name-in-event", fmt.Errorf("expected check_name to exist in event.Attributes"))
+		return
+	}
+
+	emitter.resourceChecksVec.WithLabelValues(scopeID, checkName).Inc()
```

Comment:
The definition of  `resourceChecksVec` is

```go

	resourceChecksVec := prometheus.NewCounterVec(
		prometheus.CounterOpts{
			Namespace: "concourse",
			Subsystem: "resource",
			Name:      "checks_total",
			Help:      "Counts the number of resource checks performed",
		},
		[]string{"team", "pipeline"},
```

but here it seems like we're passing `scopeID` and `checkName`, rather than `team` and `pipeline` 🤔  is this right?

Reply:
np! here's the diff for `docker-compose.yml` that I used:

```diff
diff --git a/docker-compose.yml b/docker-compose.yml
index 400598ee0..e055d324d 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -29,6 +29,9 @@ services:
       CONCOURSE_ADD_LOCAL_USER: test:test,guest:guest
       CONCOURSE_MAIN_TEAM_LOCAL_USER: test
       CONCOURSE_CLUSTER_NAME: dev
+      CONCOURSE_PROMETHEUS_BIND_IP: 0.0.0.0
+      CONCOURSE_PROMETHEUS_BIND_PORT: 9100
+      CONCOURSE_ENABLE_LIDAR: "true"
 
   worker:
     build: .
```

Reply:
Yeah in the latest commit I tested prometheus emitter with that it should be good.